### PR TITLE
Switch to SP2 for Cloud 7

### PR DESCRIPTION
**Pull request by rsalevsky**

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -993,6 +1026,20 @@ function onadmin_prepare_sles12sp1_repo()
     done
 }
 
+function onadmin_prepare_sles12sp2_repo()
+{
+    for arch in x86_64 s390x; do
+        local sles12sp2_mount="$tftpboot_suse12sp2_dir/$arch/install"
+        add_mount "SLE-12-SP2-Server-LATEST/sle-12-$arch" \
+            "$clouddata:/srv/nfs/suse-12.1/$arch/install" \
```
Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: `suse-12.2` ?

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -1423,14 +1501,22 @@ EOF
...
+                zypper ar http://${clouddata}/repos/SLES12-SP2-Updates/ sles12sp2up
+            ;;
         esac
     fi
 
-    if iscloudver 6plus ; then
+
+    if iscloudver 7plus; then
+        onadmin_prepare_sles12sp2_repos
+        onadmin_prepare_sles12plus_cloud_repos
```
Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: Don't we need SP1 repos for Cloud7 as well? (for SES). If so, we likely need to adapt also `export_tftpboot_repos_dir`

**Comment by jsuchome** [comment]
+1

File: scripts/mkcloud [hunk 3]
```diff
@@ -638,6 +620,38 @@ function macfunc()
     printf "52:54:77:77:77:%02x" $1
 }
 
+function get_admin_node_dist()
+{
+    # echo the name of the current dist for the admin node
+    local dist=
```
Comment (aplanas) [inline scripts/mkcloud]: IMHO is more clear assign here SLE11 and remove the last `else`.

**Comment by aplanas** [comment]
+1

**Comment by rsalevsky** [comment]
There is a problem while installing the nodes which needs to be fixed before merging. I'm back on Monday but feel free to take this task.

File: scripts/mkcloud [hunk 4]
```diff
@@ -638,6 +620,38 @@ function macfunc()
...
+    echo "$dist"
+}
+
+function get_admin_node_image_name()
+{
+    # get the name of the image to deploy the admin node
+    local dist=`get_admin_node_dist`
+    case $dist in
+        SLE12SP1) image=SLES12-SP1.qcow2 ;;
+        SLE12SP2) image=SLES12-SP2.qcow2 ;;
```
Comment (vuntz) [inline scripts/mkcloud]: I'd reverse the order of these two lines to be consistent with the general ordering that we have here.

File: scripts/qa_crowbarsetup.sh [hunk 5]
```diff
@@ -1423,17 +1501,23 @@ EOF
...
 
-    if iscloudver 6plus ; then
-        if [[ $cloudsource =~ ^M[1-7]$ ]]; then
-            onadmin_prepare_sles12_repos
-        fi
+
+    if iscloudver 7plus; then
+        onadmin_prepare_sles12sp1_repos
+        onadmin_prepare_sles12sp2_repos
+        onadmin_prepare_sles12plus_cloud_repos
```
Comment (vuntz) [inline scripts/qa_crowbarsetup.sh]: This call means that `onadmin_create_sles12plus_repos` needs to be updated.

File: scripts/qa_crowbarsetup.sh [hunk 6]
```diff
@@ -504,18 +504,19 @@ function resize_partition()
 
 function export_tftpboot_repos_dir()
 {
-    tftpboot_repos_dir=/srv/tftpboot/repos
-    tftpboot_suse_dir=/srv/tftpboot/suse-11.3
-
-    if iscloudver 5; then
+    if iscloudver 7plus; then
+        tftpboot_suse12sp2_dir=/srv/tftpboot/suse-12.2
+        tftpboot_repos12sp2_dir=$tftpboot_suse12sp2_dir/x86_64/repos
```
Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: I believe we need to set also`tftpboot_suse12sp1_dir` here, it is used from ` onadmin_prepare_sles12sp1_repo`

Comment (vuntz) [inline scripts/qa_crowbarsetup.sh]: You need to keep setting the `sp1_dir` variables, as the SES nodes will need that.

File: scripts/qa_crowbarsetup.sh (hunk 5 above)
Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: Ah, right, you mean `targetdir="$tftpboot_repos12sp1_dir"` in `onadmin_create_sles12plus_repos`

**Comment by vuntz** [comment]
This was not mentioned in a comment, but this got continued in https://github.com/SUSE-Cloud/automation/pull/999

**Pull request closed without merging**